### ISSUE TITLE
Cannot read property 'suggestion' of undefined

### ISSUE THREAD

**Issue by thanhthaohoang** [KazanExpress/vue-simple-suggest#62]
## **I'm submitting a ...**
- [ ] bug report
- [ ] support request
- [ ] feature request

## **What is the current behavior?**
I don't understand how to use custom suggestion items. I got this error : 
`Cannot read property 'suggestion' of undefined`

## **If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem**
Here it is my getList() method 
```
methods: {
      getList(inputValue) {
        let jsonList = this.$store.getters['places/getPlacesN1']
        let arrayList = []
        jsonList.forEach( el => {
          arrayList.push({
            id: el.fields.googlePlaceId.fr,
            name: el.fields.name.fr
          })
        })
        return arrayList
      }
    }
```

In my template 
```
<vue-simple-suggest
      v-model="query"
      :list="getList"
      placeholder="Essayer 'Planétarium'"
      type="search"
      ref="suggestComponent"
      mode="input"
      :filter-by-query="false"
      >

      <div slot="suggestion-item" slot-scope="scope" :title="scope.suggestion.name">
          <div class="text">
            <span v-html="boldenSuggestion(scope)"></span>
          </div>
      </div>
</vue-simple-suggest>

```
## **What is the expected behavior?**
I want to display a specific list reached from json in suggestions items. However, I have 'undefined' written in it. 

## **Please tell us about your environment:**
  
  - Version: 2.0.0-beta.X
  - Browser: Chrome 66
  - Language: ES6

**Comment by Raiondesu** [comment]
Hello, @thanhthaohoang. :)
Are you using the `boldenSuggestion` method from the example?

Can you, please, provide any error stack traces?

**Comment by thanhthaohoang** [comment]
Hi @Raiondesu ! Thanks for your quick response.

Yes, I'm using boldenSuggestion method but it seems that scope doesn't exist. 
Here a screenshot of the error 
![image](https://user-images.githubusercontent.com/23716648/40125143-7882ffa6-592a-11e8-8f7d-f218cae525b0.png)

**Comment by Raiondesu** [comment]
Maybe like so:
```js
boldenSuggestion(scope) {
  if (!scope) return '';

  const { suggestion, query } = scope;

  let result = this.$refs.suggestComponent.displayProperty(suggestion);

  if (!query) return result;

  const texts = query.split(/[\s-_/\\|\.]/gm).filter(t => !!t) || [''];
  return result.replace(new RegExp('(.*?)(' + texts.join('|') + ')(.*?)','gi'), '$1<b>$2</b>$3');
}
```

**Comment by Raiondesu** [comment]
@thanhthaohoang, please, let me know if this helps.

**Comment by thanhthaohoang** [comment]
I checked my resulting lists and there is no undefined elements.
Unfortunately, your solution doesn't work. There is nothing written in suggestions item and I have those warnings : 
`[Vue warn]: Property or method "scope" is not defined on the instance but referenced during render. Make sure to declare reactive data properties in the data option.`
`[Vue warn]: Duplicate presence of slot "suggestion-item" found in the same render tree - this will likely cause render errors.`

**Comment by Raiondesu** [comment]
@thanhthaohoang, you may also want to specify a custom `displayAttribute` in your template, because your current attribute (`name`) does not correspond with component's defaults.
Like so:
```html
<vue-simple-suggest
  v-model="query"
  :list="getList"
  placeholder="Essayer 'Planétarium'"
  type="search"
  ref="suggestComponent"
  display-attribute="name"
>
  <div slot="suggestion-item" slot-scope="scope" :title="scope.suggestion.name">
    <div class="text">
      <span v-html="boldenSuggestion(scope)"></span>
    </div>
  </div>
</vue-simple-suggest>
```

**Comment by Raiondesu** [comment]
I didn't quite understand the part about using json as the resulting list.
The format is specified in the [docs](https://github.com/KazanExpress/vue-simple-suggest#props).
`list`:
> The array provider function, must accept a query as its only argument. Can return an array or a promise. Can be async. The component behaves as a simple input without this function.

Anything other than array is non-conventional in this case.

I would suggest you analyze the error stack trace carefully and determine exactly where the error occurs.
From your description I understood that your list's format is not quite right.

In the meantime, I'll try to prevent the `undefined` errors by handling them in a softer manner.

**Comment by Raiondesu** [comment]
Anyway, the issue seems to be not a bug in `vue-simple-suggest`, but rather a bug in the usage code.
I will close the issue until proven otherwise. A context-independent reproduction (via jsfiddle, for example) would suit this purpose perfectly.

**Comment by kaskar2008** [comment]
I think I see what's the problem :)
About your Vue's version: the latest version released is `2.5.16`.
@thanhthaohoang You should try to use our component like this:
```html
<vue-simple-suggest
      v-model="query"
      :list="getList"
      placeholder="Essayer 'Planétarium'"
      type="search"
      ref="suggestComponent"
      mode="input"
      value-attribute="fields.googlePlaceId.fr"
      display-attribute="fields.name.fr"
      :filter-by-query="false"
      >

      <div slot="suggestion-item" slot-scope="{ suggestion, query }">
          <div class="text">
            <span v-html="boldenSuggestion({ suggestion, query })"></span>
          </div>
      </div>
</vue-simple-suggest>

<script>
import VueSimpleSuggest from 'vue-simple-suggest'
import 'vue-simple-suggest/dist/styles.css' // Optional CSS

  export default {
    components: {
      VueSimpleSuggest
    },
    data () {
      return {
        query: '',
        loading: false
      }
    },
    methods: {
      boldenSuggestion(scope) {
        if (!scope) return '';

        const { suggestion, query } = scope;

        let result = this.$refs.suggestComponent.displayProperty(suggestion);

        if (!query) return result;

        const texts = query.split(/[\s-_/\\|\.]/gm).filter(t => !!t) || [''];
        return result.replace(new RegExp('(.*?)(' + texts.join('|') + ')(.*?)','gi'), '$1<b>$2</b>$3');
      },
      getList(inputValue) {
        return this.$store.getters['places/getPlacesN1']
      }
    }
  }
</script>
```

**Comment by kaskar2008** [comment]
@thanhthaohoang Can you provide us your entire code via jsbin/jsfiddle or any other sandboxes? I think you've made an error somewhere else.
